### PR TITLE
[@mantine/core] Tab: set to add aria-controls with panelId conditiona…

### PR DESCRIPTION
**Pull request by forestileao**
…lly to the panel existance

Issue: https://github.com/mantinedev/mantine/issues/4141

Added verification inside Tab.tsx to check if the pointing Panel exists in the DOM. Else it sets to undefined.

### CODE REVIEW

File: src/mantine-core/src/Tabs/Tab/Tab.tsx [hunk 1]
```diff
@@ -54,6 +54,8 @@ export const Tab = forwardRef<HTMLButtonElement, TabProps>((props, ref) => {
   );
 
   const isActive = value === ctx.value;
+  const panelId = ctx.getPanelId(value);
+  const ariaControlsValue = document.getElementById(panelId) ? panelId : undefined;
```
Comment (rtivital) [inline src/mantine-core/src/Tabs/Tab/Tab.tsx:58]: This will throw an error during server side rendering.

Comment (rtivital) [reply src/mantine-core/src/Tabs/Tab/Tab.tsx:58]: You can use ModalBase component as reference, it has similar logic

Comment (forestileao) [reply src/mantine-core/src/Tabs/Tab/Tab.tsx:58]: I will take a look, thanks

Comment (forestileao) [reply src/mantine-core/src/Tabs/Tab/Tab.tsx]: @rtivital I added a mounted panel list to its provider, so document.getElementById is not necessary now. Based on your recommendation of ModalBase

Comment (forestileao) [reply src/mantine-core/src/Tabs/Tab/Tab.tsx]: Fixed eslint errors after test.

**Comment by rtivital** [comment]
Thanks!

**Pull request merged**